packages/react-native/src/session.ts

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+import type { StoreApi, UseBoundStore } from 'zustand'
+import { createWorldBridgeStore } from '@worldcoin/idkit-core'
+import type {
+	WorldBridgeStore,
+	IDKitConfig,
+	VerificationState,
+	ISuccessResult,
+	AppErrorCodes,
+} from '@worldcoin/idkit-core'
+
+export type SessionStatus = {
+	state: VerificationState
+	result: ISuccessResult | null
+	errorCode: AppErrorCodes | null
+}
+
+export class Session {
+	private store: UseBoundStore<StoreApi<WorldBridgeStore>>
+
+	constructor() {
+		this.store = createWorldBridgeStore()
+	}
+
+	async create(
+		app_id: IDKitConfig['app_id'],
+		action: IDKitConfig['action'],
+		options?: {
+			signal?: IDKitConfig['signal']
+			bridge_url?: IDKitConfig['bridge_url']
+			verification_level?: IDKitConfig['verification_level']
+			action_description?: IDKitConfig['action_description']
+			partner?: IDKitConfig['partner']
+		}
+	): Promise<Session> {
+		await this.store.getState().createClient({
+			app_id,
+			action,
+			signal: options?.signal,
+			bridge_url: options?.bridge_url,
+			action_description: options?.action_description,
+			verification_level: options?.verification_level,
+			partner: options?.partner,
+		})
+		return this
+	}
+
+	/**
+	 * Polls for verification updates
+	 */
+	public async pollForUpdates(): Promise<void> {
+		return this.store.getState().pollForUpdates()
+	}
+
+	/**
+	 * Polls for updates and returns the current session status
+	 */
+	public async status(): Promise<SessionStatus> {
+		await this.pollForUpdates()
+		const state = this.store.getState()
+
+		return {
+			state: state.verificationState,
+			result: state.result,
+			errorCode: state.errorCode,
+		}
+	}
+
+	/**
+	 * Gets the current connector URI that can be used to start verification
+	 */
+	public get connectorURI(): string | null {
```

Comment:
tbh while doing this worth considering renaming some stuff. I don't think it was original thought through very deeply. 

SessionURI for instance might be more clear here. 
```suggestion
	public get sessionURI(): string | null {
```